### PR TITLE
Update gravity form to support ConnectWise API version 2020.1 

### PR DESCRIPTION
**Pull request by saowaluck**
Changes proposed in this pull request:
- Support ConnectWise API version 2020.1 by change
  - Add `client_id` settings field and send value to the header
  - Change the structure of creating a company and contact field
Screen Shot 2563-02-18 at 22 33 00

   - Result when test create 
     - Create New Opportunity, Activity, Service Ticket
Screen Shot 2563-02-19 at 08 39 21

[See more](https://staging.connectwisedev.com/v2020_2/ConnectWise.aspx?locale=en_US&session=new#XQAACABlAgAAAAAAAAA9iIoG07$U9XlKo4yI1rb2q$OhmbypMeK$QqSW9Vty0rWCseu$iUhqMzJNxv6AIDS9vmocicvQcPcmX44xzRRGcNFHveO0A$X$bCUKVspvOymOl46X_wX7GI01SDNRBX4kqfHp7XwAkoZkGL_3WN5MCJEDcKz2oYhiZW6Tpp5NzN5pJquQkZtoaiFavLa4SgmAP4CghiCfce38fcqAoiWxdGFe5zKCUJnodjchSTseD3Bb128_3GIdtUs4ZaX4l50LU9ObBiunZrLGIAlfpuv_H8ZjwOmmOKIyNOySff_BVU60_He5imd4Yv$BFEQA??CompanyDetail)

Connected to https://trello.com/c/P7EDfZib/5581-feature-request-upgrading-gravity-forms-connectwise-add-on-plugin

### CODE REVIEW

**Comment by gatukgl** [comment]
@saowaluck Is there any card related to this?

File: class-gf-connectwise-2020v1.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		$this->log_debug( __METHOD__ . '(): start getting team members from ConnectWise' );
+
+		$team_members_list    = array();
+		$get_team_members_url = 'system/members?pageSize=200';
+		$cw_team_members      = $this->send_request( $get_team_members_url, 'GET', NULL );
+		$cw_team_members      = json_decode( $cw_team_members['body'] );
+
+		foreach ( $cw_team_members as $each_member ) {
+			$member = array(
+					'label' => esc_html__( $each_member->firstName . ' ' . $each_member->lastName, 'gravityformsconnectwise' ),
```
Comment (gatukgl) [inline class-gf-connectwise-2020v1.php]: Is this need to indent like this -0-. It's so far.

File: class-gf-connectwise-2020v1.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	public static function get_instance() {
+		if ( null == self::$_instance ) {
+			self::$_instance = new self;
+		}
+
+		return self::$_instance;
+	}
+
+	public function get_team_members() {
+		$this->log_debug( __METHOD__ . '(): start getting team members from ConnectWise' );
```
Comment (gatukgl) [inline class-gf-connectwise-2020v1.php]: Just wonder, where does this log goes to? In the console?

File: class-gf-connectwise.php [hunk 3]
```diff
@@ -1417,9 +1425,10 @@ public function send_request( $url, $request_method, $body, $error_notification
 			$this->log_debug( __METHOD__ . '(): body => ' . json_encode( $body, JSON_PRETTY_PRINT ) );
 		}
 
-		$company_id  = $this->get_plugin_setting( 'company_id' );
-		$public_key  = $this->get_plugin_setting( 'public_key' );
-		$private_key = $this->get_plugin_setting( 'private_key' );
+		$company_id        = $this->get_plugin_setting( 'company_id' );
+		$public_key        = $this->get_plugin_setting( 'public_key' );
+		$private_key       = $this->get_plugin_setting( 'private_key' );
+        $client_id         = $this->get_plugin_setting( 'client_id' );
```
Comment (gatukgl) [inline class-gf-connectwise.php]: Seems like this one is not like its friend.

File: tests/GravityFormsConnectWiseTest.php [hunk 4]
```diff
@@ -80,6 +80,13 @@ function test_addon_settings_should_have_fields() {
 		$this->assertTrue( array_key_exists( "save_callback", $actual[0]["fields"][3] ) );
 		$this->assertTrue( array_key_exists( "feedback_callback", $actual[0]["fields"][3] ) );
 
+		$this->assertEquals( $actual[0]["fields"][4]["name"], "client_id" );
+		$this->assertEquals( $actual[0]["fields"][4]["label"], "Client ID" );
+		$this->assertEquals( $actual[0]["fields"][4]["type"], "text" );
+		$this->assertEquals( $actual[0]["fields"][4]["class"], "small" );
```
Comment (gatukgl) [inline tests/GravityFormsConnectWiseTest.php]: I just saw sometimes we use single quote and sometimes double quote. Which one that we really use?

File: class-gf-connectwise-2020v1.php (hunk 2 above)
Comment (saowaluck) [reply class-gf-connectwise-2020v1.php]: Screen Shot 2563-02-19 at 15 15 20

File: tests/GravityFormsConnectWiseTest.php (hunk 4 above)
Comment (saowaluck) [reply tests/GravityFormsConnectWiseTest.php]: It uses one single quote 
but I  change related with code before

File: class-gf-connectwise-2020v1.php (hunk 2 above)
Comment (saowaluck) [reply class-gf-connectwise-2020v1.php]: it show in the log file
Screen Shot 2563-02-19 at 15 15 26

**Pull request merged**